File: Back-end/AutionEaseProject/src/main/java/com/trade/autioneaseproject/entity/AssetType.java
package com.trade.autioneaseproject.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "AssetTypes")
public class AssetType {
    @Id
    private Integer assetTypeID;
    private String assetTypeName;
    private Boolean delflag;
}

File: Back-end/AutionEase_Laravel/app/Models/AssetType.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
// use Illuminate\Database\Eloquent\SoftDeletes;

class AssetType extends Model
{
    use HasFactory;

    protected $table = 'AssetTypes'; // Chỉ định tên bảng nếu cần
    protected $primaryKey = 'assetTypeID'; // Đặt primary key cho model
    public $incrementing = false; // Nếu primary key không tự động tăng
    protected $keyType = 'string'; // Đặt kiểu của primary key nếu nó không phải là integer

    protected $fillable = [
        'assetTypeID',
        'assetTypeName',
        'delflag', // Sửa lỗi đánh máy ở đây
    ];
}
